src/obs_inv_utils/inventory_table_factory.py

```diff
@@ -376,33 +466,59 @@ def insert_obs_inv_items(obs_inv_items):
     for obs_item in obs_inv_items:
         if not isinstance(obs_item, se.TarballFileMeta):
             msg = 'Each observation inventory item must be in the form' \
-                  f' of TarballFileMeta. Item type: {type(obs_item)}'
+                f' of TarballFileMeta. Item type: {type(obs_item)}'
             raise TypeError(msg)
 
-        tbl_item = ObsInventory(
-            cmd_result_id=obs_item.cmd_result_id,
-            filename=obs_item.filename,
-            parent_dir=obs_item.parent_dir,
-            platform=obs_item.platform,
-            s3_bucket=obs_item.s3_bucket,
-            prefix=obs_item.prefix,
-            cycle_tag=obs_item.cycle_tag,
-            data_type=obs_item.data_type,
-            cycle_time=obs_item.cycle_time,
-            obs_day=obs_item.obs_day,
-            data_format=obs_item.data_format,
-            suffix=obs_item.suffix,
-            nr_tag=obs_item.nr_tag,
-            file_size=obs_item.file_size,
-            etag=obs_item.etag,
-            permissions=obs_item.permissions,
-            last_modified=obs_item.last_modified,
-            inserted_at=obs_item.inserted_at,
+        hash_value = generate_obs_inventory_hash(obs_item.filename, obs_item.parent_dir, obs_item.platform,
+                                                 obs_item.s3_bucket)
+
+        row = {
+            'cmd_result_id': obs_item.cmd_result_id,
+            'filename': obs_item.filename,
+            'parent_dir': obs_item.parent_dir,
+            'platform': obs_item.platform,
+            's3_bucket': obs_item.s3_bucket,
+            'prefix': obs_item.prefix,
+            'cycle_tag': obs_item.cycle_tag,
+            'data_type': obs_item.data_type,
+            'cycle_time': obs_item.cycle_time,
+            'obs_day': obs_item.obs_day,
+            'data_format': obs_item.data_format,
+            'suffix': obs_item.suffix,
+            'nr_tag': obs_item.nr_tag,
+            'file_size': obs_item.file_size,
+            'etag': obs_item.etag,
+            'permissions': obs_item.permissions,
+            'last_modified': obs_item.last_modified,
+            'unique_hash': hash_value,
+            'inserted_at': obs_item.inserted_at,
+            'valid_at': obs_item.valid_at,
+        }
+        rows.append(row)
+
+    #handle the best way available for each database type
+    if(database_type.lower() == 'mysql'):
+        statement = mysql_insert(ObsInventory).values(rows)
+        statement = statement.on_duplicate_key_update(
+            valid_at=statement.inserted.valid_at
+        )
+    else:
+        #sqlite specific
+        statement = sqlite_insert(ObsInventory).values(rows)
+        statement = statement.on_conflict_do_update(
+            index_elements=['filename',
```

Comment:
index_elements list does not match the unique constraint defined above. Should be: index_elements=['unique_hash',
            'obs_day',
            'file_size',
            'last_modified',
            'etag'],

Reply:
great catch! updated!